mcsema/CFG/CFG.cpp

```diff
@@ -163,39 +152,60 @@ static bool ResolveReference(NativeModule *module, NativeXref *xref) {
     }
   }
 
-  // Try to recover by finding a local variable.
+  if (xref->target_segment) {
+    return true;
+  }
+
+  // Try to recover by finding a global variable.
   for (const auto &entry : module->ea_to_var) {
-    auto var = entry.second;
-    if (var->name == xref->target_name) {
-      auto final_var = reinterpret_cast<const NativeVariable *>(var->Get());
-      xref->var = final_var;
-      xref->target_segment = final_var->segment;
-      module->ea_to_var[xref->target_ea] = final_var;
-
-      LOG(ERROR)
-          << "Attempting reference fix at " << std::hex << xref->ea
-          << " targeting " << xref->target_name << " using variable "
-          << std::hex << var->ea << ", resolving to variable "
-          << final_var->name << " at " << std::hex << final_var->ea;
-      return true;
+    const auto &var = entry.second;
+    if (var->name != xref->target_name) {
+      continue;
     }
+
+    const auto new_var = new NativeVariable;
+    const auto final_var = reinterpret_cast<const NativeVariable *>(var->Get());
+    new_var->forward = var->Get();
+    new_var->ea = xref->target_ea;
+    new_var->name = xref->target_name;
+    new_var->segment = final_var->segment;
```

Comment:
Maybe we want some constructor of `NativeVariable` that takes things on line 168 - 171?

Reply:
In spirit I agree with this. But then I am just going to wait for the update that changes everything away from CFG files :-P